bitcoin/src/crypto/sighash.rs

```diff
@@ -45,11 +45,11 @@ macro_rules! impl_message_from_hash {
 hash_newtype! {
     /// Hash of a transaction according to the legacy signature algorithm.
     #[hash_newtype(forward)]
-    pub struct LegacySighash(sha256d::Hash);
+    pub struct LegacySighash(pub sha256d);
 
     /// Hash of a transaction according to the segwit version 0 signature algorithm.
     #[hash_newtype(forward)]
-    pub struct SegwitV0Sighash(sha256d::Hash);
+    pub struct SegwitV0Sighash(pub sha256d);
```

Comment:
I guess this makes sense but why not publish the field of all hashes?

Reply:
The inner field gives access to the public `engine` and `from_engine` functions which we would like to avoid for what I've called in the new example code a "limited hash".

hmm, uncovered another bug - please see `QUESTION` stuff in `examples/hash_newtype.rs`

Reply:
visibility stuff has been totally re-written now.

Reply:
> The inner field gives access to the public `engine` and `from_engine` functions which we would like to avoid

My idea was to make custom-hashing discouraged not disallow it completely. It can still be bypassed using `from_byte_array` which is fine.

Reply:
This might be a better approach indeed. Not sure.